### ISSUE TITLE
`numpy.typeDict` is deprecated

### ISSUE THREAD

**Issue by otizonaizit** [mdp-toolkit/mdp-toolkit#92]
In https://github.com/mdp-toolkit/mdp-toolkit/blob/300d288d75ac8495f7f4a0ef40cb71d4ff58a311/mdp/utils/routines.py#L295
we use `np.typeDict`. This is deprecated since forever: https://github.com/numpy/numpy/commit/668950285c407593a368336ff2e737c5da84af7d and https://github.com/numpy/numpy/issues/17585 . Since numpy release [1.21](https://github.com/numpy/numpy/releases/tag/v1.21.0) this now issues a deprecation warning: https://github.com/numpy/numpy/pull/17586
It seems that `scipy` is not importing deprecated names from the numpy namespace anymore, so that with numpy 1.21 and scipy 1.7 you now get an error:

```python
AttributeError: module 'scipy' has no attribute 'typeDict'
```

We should change `np.typeDict` with `np.sctypeDict` most probably, but it is probably better to try to understand what we were trying to do in the function `get_dtypes` and fix it properly:
https://github.com/mdp-toolkit/mdp-toolkit/blob/300d288d75ac8495f7f4a0ef40cb71d4ff58a311/mdp/utils/routines.py#L274

Thanks to Elgin Road for reporting this!

**Comment by Stewori** [comment]
Thank you, Nils.

**Comment by finnoshea** [comment]
Using conda and a clean install of mdp with `conda install mdp` I can `import mdp` without problem.  But, if I then install scipy (and do nothing else), when I `import mdp` I get the AttributeError mentioned in the op of this issue.  Is there a work around for using both mdp and scipy?

**Comment by NiMlr** [comment]
@finnoshea Do you experience the same problem when you install the mdp version from the master branch:
```bash
pip install git+https://github.com/mdp-toolkit/mdp-toolkit
```

**Comment by finnoshea** [comment]
That does appear to work.  To recap:

0) new conda env
1) conda install mdp -> import mdp works
2) conda install scipy -> import mdp fails

versus

0) new conda env
1) pip install git+https://github.com/mdp-toolkit/mdp-toolkit -> import mdp works
2) conda install scipy -> import mdp still works

Thanks for the pointer @NiMlr!

**Comment by Stewori** [comment]
This issue has been fixed in #94, so I am closing it now.
